pkg/token/federatedIdentity.go

```diff
@@ -0,0 +1,94 @@
+package token
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+)
+
+type workloadIdentityToken struct {
+	clientID           string
+	tenantID           string
+	federatedTokenFile string
+	authorityHost      string
+	resourceID         string
+}
+
+func newWorkloadIdentityToken(clientID, federatedTokenFile, authorityHost, resourceID, tenantID string) (TokenProvider, error) {
+	if clientID == "" {
+		return nil, errors.New("clientID cannot be empty")
+	}
+	if tenantID == "" {
+		return nil, errors.New("tenantID cannot be empty")
+	}
+	if federatedTokenFile == "" {
+		return nil, errors.New("federatedTokenFile cannot be empty")
+	}
+	if authorityHost == "" {
+		return nil, errors.New("authorityHost cannot be empty")
+	}
+	if resourceID == "" {
```

Comment:
i think we should rename `resourceID` to `serverID` as that's what is being used in the input.